### PR TITLE
feat(techdocs-container): pre config repo_url based on catalog-info

### LINKED ISSUE #2903: Pre-configure techdocs sites with repo_url in mkdocs.yml file
To show the edit button on pages of documentation the repo_url config need to be set in the mkdocs.yml file. To avoid one extra step of configuration we want this to be pre-configured in the process of generating documentation. 

The location of the repository can be gathered from the techdocs_ref annotation as it is now, even though this might change in the future. 

The edit button is the "pen" to the right.
Screen Shot 2020-10-14 at 5 18 57 PM

### PR DESCRIPTION
**Pull request by Marvin9**
## Hey, I just made a Pull Request!

Resolves: #2903 

### Summary:

Before running ```mkdocs```, docker will configure ```repo_url``` in ```mkdocs.yml``` based on ```techdocs-ref``` value in ```catalog-info```. This step will be skipped otherwise.

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/spotify/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)

### CODE REVIEW

**Comment by emmaindal** [comment]
Hey @Marvin9 thanks a lot for this contribution. Two things:

1. The catalog-info.yaml will actually not always be in the same dir as the mkdocs.yaml file, and especially not with the changes we are about to make soon. Therefore we would like to be able to pass in a repo_url to the container in the generator step as a parameter, we can get this from the Entity which we have access to in the preparer so pass that through to the generator would solve it.

2. Followed by the 1 point - You could keep whats there with getting it from the catalog-info, if a catalog-info exist in the same dir (for use cases where not our backend is used but rather the container separatly), but fall back to the repo_url parameter that are passed to the container in the generator.

Let us know if you have any follow-up questions. We can also jump into a mob-session to solve this together if you want! 🥳

File: packages/techdocs-container/Dockerfile [hunk 1]
```diff
@@ -18,11 +18,13 @@ FROM python:3.8-alpine
 RUN apk update && apk --no-cache add gcc musl-dev openjdk11-jdk curl graphviz ttf-dejavu fontconfig
 
 RUN curl -o plantuml.jar -L http://sourceforge.net/projects/plantuml/files/plantuml.1.2020.16.jar/download && echo "c789ace48347c43073232b1458badc5810c01fe8  plantuml.jar" | sha1sum -c - && mv plantuml.jar /opt/plantuml.jar
-RUN pip install --upgrade pip && pip install mkdocs-techdocs-core==0.0.8
+RUN pip install --upgrade pip && pip install mkdocs-techdocs-core==0.0.8 && pip install shyaml
```
Comment (hooloovooo) [inline packages/techdocs-container/Dockerfile:21]:
```suggestion
RUN pip install --upgrade pip && pip install mkdocs-techdocs-core==0.0.8 shyaml==0.5.1
```

**Comment by OrkoHunter** [comment]
Hey @Marvin9 ! I hope you are doing well. I wonder if you are interested in continuing this work or would like us to take it from here? :)

**Comment by Marvin9** [comment]
@OrkoHunter sure. Please take it up.

**Comment by OrkoHunter** [comment]
Appreciate your quick response @Marvin9.

**Comment by andrewthauer** [comment]
@OrkoHunter - A while back I started working extending the CLI to auto generate the entire mkdoc.yml file automatically. As I've mentioned in the past, we currently only use the reader component and have a custom docs site build. However I'd like to start migrating our custom docs build to techdocs.

I was thinking for us it would be easier to do migrate parts into the CLI first and then eventually get to using the backend. I'm wondering if having a shared techdocs package between the CLI and backend could be a thing. What I started working on (but have since gotten sidetracked) could be used in both the CLI and backend to build docs sites.

**Comment by OrkoHunter** [comment]
> I'm wondering if having a shared techdocs package between the CLI and backend could be a thing.

Yes, you are spot on! We feel the need for a `techdocs-common` package as well. With the new [cloud storage features](https://github.com/backstage/backstage/pull/3336) in techdocs, we want to share these external integrations between CLI (techdocs-cli) and techdocs plugins.

**Pull request closed without merging**